class.jetpack.php

```diff
@@ -4019,6 +4022,45 @@ function plugin_action_links( $actions ) {
 		return array_merge( $jetpack_home, $actions );
 	}
 
+	/**
+	 * Filters the login URL to include the registration flow in case the user isn't logged in.
+	 *
+	 * @param string $login_url The wp-login URL.
+	 * @param string $redirect  URL to redirect users after logging in.
+	 * @since Jetpack 8.4
+	 * @return string
+	 */
+	public function login_url( $login_url, $redirect ) {
+		parse_str( wp_parse_url( $redirect, PHP_URL_QUERY ), $redirect_parts );
+		if ( ! empty( $redirect_parts['connect_login_redirect'] ) ) {
```

Comment:
Lets make `connect_login_redirect` a class constant and prefix it with `jetpack_` this way it has less chance of conflicting with other plugins. 
By making it a class constant the string is easier to check that it is the same one.

Reply:
Resolved in 37c6a53196597e9353fe86861f5b9324a32595ff.